### DISCUSSION TITLE
How to show all diffs from the previous invocation of webchanges?

### DISCUSSION THREAD

**Discussion by Markus00000** [mborsetti/webchanges#80] [Q&A]
If I accidentally lose the diff output after running webchanges, can it be shown again? What I did in the past was to roll back the database and run webchanges again. If I don’t know which jobs found changes, using ``--test-differ`` for all jobs would take a while.

Just wondering if there is a better way or if this would be a feature request.

**Comment by mborsetti** [comment]
Sorry to hear this!

I can't think of a better way than using --test-differ.

I'll noodle on the feature request.

**Reply by Markus00000** [reply]
What do you think about the following:

* Store each snapshot’s creation timestamp.
* List snapshots chronologically.
* Show all diffs starting at snapshot n.

Example usage:

    $ webchanges --list-snapshots 4
    4: 2024-04-17 08:39:50 Job 3
    3: 2024-04-23 10:28:15 Job 1
    2: 2024-04-23 10:28:16 Job 2
    1: 2024-04-25 09:08:42 Job 1
    $ webchanges --diff-snapshots 3
    # show 2 diffs for job 1 and 1 diff for job 2

Commands could be:

    --list-snapshots [NUM_SNAPSHOTS]
    --diff-snapshots NUM_SNAPSHOTS

``--list-snapshots`` could use the same format as ``--list`` but insert the date between index and job name.

By default these commands could operate on all jobs with the option to only show a specific one:

    --list-snapshots NUM_SNAPSHOTS [JOB]
    --diff-snapshots NUM_SNAPSHOTS [JOB]

``--list-snapshots`` might benefit from a sensible default number of snapshots or make the argument required.

**Edit:**  Just noticed that ``--diff-snapshots`` would be ``--test-differ`` applied to all jobs/snapshots.
